cmd/build.go

```diff
@@ -424,21 +431,99 @@ func buildNormal(targetOS string, vmArguments []string) {
 			}
 		}
 	}
+	if !buildOmitFlutterBundle {
+		var trackWidgetCreation string
+		if buildMode == build.DebugMode {
+			trackWidgetCreation = "--track-widget-creation"
+		}
 
-	outputEngineFile := filepath.Join(build.OutputDirectoryPath(targetOS), build.EngineFile(targetOS))
-	err = copy.Copy(
-		filepath.Join(engineCachePath, build.EngineFile(targetOS)),
-		outputEngineFile,
-	)
-	if err != nil {
-		log.Errorf("Failed to copy %s: %v", build.EngineFile(targetOS), err)
-		os.Exit(1)
-	}
+		cmdFlutterBuild := exec.Command(build.FlutterBin, "build", "bundle",
+			"--asset-dir", filepath.Join(build.OutputDirectoryPath(targetOS), "flutter_assets"),
+			"--target", buildTarget,
+			trackWidgetCreation,
+		)
+		cmdFlutterBuild.Stderr = os.Stderr
+		cmdFlutterBuild.Stdout = os.Stdout
 
-	err = copy.Copy(
-		filepath.Join(engineCachePath, "artifacts", "icudtl.dat"),
-		filepath.Join(build.OutputDirectoryPath(targetOS), "icudtl.dat"),
-	)
+		log.Infof("Bundling flutter app")
+		err = cmdFlutterBuild.Run()
+		if err != nil {
+			log.Errorf("Flutter build failed: %v", err)
+			os.Exit(1)
+		}
+		if buildMode.IsAot {
+			err := os.Remove(filepath.Join(build.OutputDirectoryPath(targetOS), "flutter_assets", "isolate_snapshot_data"))
+			if err != nil {
+				log.Errorf("Failed to remove unused isolate_snapshot_data: %v", err)
+				os.Exit(1)
+			}
+			err = os.Remove(filepath.Join(build.OutputDirectoryPath(targetOS), "flutter_assets", "vm_snapshot_data"))
+			if err != nil {
+				log.Errorf("Failed to remove unused vm_snapshot_data: %v", err)
+				os.Exit(1)
+			}
+			err = os.Remove(filepath.Join(build.OutputDirectoryPath(targetOS), "flutter_assets", "kernel_blob.bin"))
+			if err != nil {
+				log.Errorf("Failed to remove unused kernel_blob.bin: %v", err)
+				os.Exit(1)
+			}
+			dart := filepath.Join(engineCachePath, "dart"+build.ExecutableExtension(targetOS))
+			genSnapshot := filepath.Join(engineCachePath, "gen_snapshot"+build.ExecutableExtension(targetOS))
+			kernelSnapshot := filepath.Join(build.OutputDirectoryPath(targetOS), "kernel_snapshot.dill")
+			elfSnapshot := filepath.Join(build.OutputDirectoryPath(targetOS), "libapp.so")
```

Comment:
I'm guessing .so because we are on linux. Maybe add a `TODO` saying we need to support other platform?

Reply:
It's always an .so. They are used platform indpendent and are only the way of how the snapshots are bundled